int32 count
---
bool finish
